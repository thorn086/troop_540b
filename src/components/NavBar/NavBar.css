
.navigation__item{
margin:2rem;
}

.navigation__checkbox{
display: none;
}
.navigation__checkbox:checked ~ .navigation__backgroud{
transform: scale(300);
}
.navigation__checkbox:checked ~ .navigation__nav{
    opacity: 1;
    width: 100%;
    }
.navigation__button{
    margin: 0;
    border: 1px solid black;
background-color:  white;
height: 3rem;
width: 3rem;
position: fixed;
top: 1.5rem;
right: 1.5rem;
border-radius: 50%;
z-index: 2000;
box-shadow: 0 1rem 3rem black;
text-align: center;
cursor: pointer;
}

.logout-btn, .home-btn, .login-btn, .signup-btn{
    font-size: 2rem; 
    font-weight: 300;
    text-transform: uppercase; 
    background-image: linear-gradient(120deg, black 80%, rgb(40, 38, 38) 5%, white 92%);
    border-radius: 15%; 
    border: 2px solid black;
    box-shadow: 0 1rem 3rem black; 
    background-size: 100%; 
    
}

.navigation__backgroud{
height: 2rem;
width: 2rem;
border-radius: 50%;
position: fixed;
top: 1.75rem;
right: 1.75rem;
background-image: radial-gradient( #c11b76,  white);
z-index: 1000;
transition: transform .8s  cubic-bezier(0.86, 0, 0.07, 1);
}

.navigation__nav{
height: 100vh;
width: 0;
position: fixed;
top: 0;
right: 0;
z-index: 1500;
opacity: 0;
transition: all .8s cubic-bezier(0.68, -0.55, 0.265, 1.55);
}

.navigation__list{
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    list-style:none ;
    text-align: center;
    color: #4c4cf2;
    width: 100%;
}
.navigation__icon{
    position: relative;
    margin-top: 1.4rem;
}
.navigation__icon, .navigation__icon::before, .navigation__icon::after{
    width: 1.25rem;
    height: 2px;
    background-color:black;
    display: inline-block;
}

.navigation__icon::before, .navigation__icon::after{
    content: '';
    position: absolute;
    left: 0;
    transition: all .2s;
    
}
.navigation__icon::before{
top:-.5rem;
}
.navigation__icon::after{
top: .5rem;
}

.navigation__button:hover .navigation__icon::before{
top: -1rem;
}
.navigation__button:hover .navigation__icon::after{
    top: 1rem;
}

.navigation__checkbox:checked + .navigation__button .navigation__icon{
    background-color: transparent;
}

.navigation__checkbox:checked + .navigation__button .navigation__icon::before{
    top: 0;
    transform: rotate(135deg);
}

.navigation__checkbox:checked + .navigation__button .navigation__icon::after{
    top: 0;
    transform: rotate(-135deg);
}