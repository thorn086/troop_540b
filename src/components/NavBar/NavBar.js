import React from 'react'
import './NavBar.css'
import { Link } from 'react-router-dom'
// import TokenService from '../../services/token-services'

function NavBar() {
    // handleLogoutClick = () => {
    //     TokenService.clearAuthToken()
    //     window.location.reload();
    // }

    // renderLogoutLink() {
    //     return (
    //         <div className='navigation__item'>
    //             <li className='navigation__item' onClick={this.handleLogoutClick}>
    //                 <button className='red-btn logout-btn'>
    //                     <Link to='/' style={{ textDecoration: 'none', color: '#fff290', padding: '1rem',fontFamily: 'Lemonada, cursive' }}> Logout</Link>
    //                 </button>
    //             </li>
    //             <li className='navigation__item'>
    //                 <button className='red-btn home-btn'>
    //                     <Link to='/' style={{ textDecoration: 'none', color: '#fff290', padding: '1rem',fontFamily: 'Lemonada, cursive' }}>Home</Link>
    //                 </button>
    //             </li>
    //         </div>
    //     )
    // }
    // toggleLoginLink() {
    //     return (
        
    //     )
    // }
    
        return (
            <div className='navigation'>
                <input type='checkbox' className='navigation__checkbox' id='navi-toggle' />
                <label htmlFor='navi-toggle' className='navigation__button'>
                    <span className='navigation__icon'></span>
                </label>
                <div className='navigation__backgroud'>&nbsp;</div>

                <nav className='navigation__nav'>
                    <ul className='navigation__list'>
                        <div className='navigation__item'>
                            <li className='navigation__item'>
                                <button className='red-button login-btn'>
                                    <Link to='/About' style={{ textDecoration: 'none', color: '#fff290', padding: '1rem', fontFamily: 'Lemonada, cursive' }}>About</Link>
                                </button>
                            </li>
                            <li className='navigation__item'>
                                <button className='red-button signup-btn'>
                                    <Link to='/Events' style={{ textDecoration: 'none', color: '#fff290', padding: '1rem', fontFamily: 'Lemonada, cursive' }}>Events</Link>
                                </button>
                            </li>
                            <li className='navigation__item'>
                                <button className='red-button home-btn'>
                                    <Link to='/Resources' style={{ textDecoration: 'none', color: ' #fff290', padding: '1rem', fontFamily: 'Lemonada, cursive' }}>Resources</Link>
                                </button>
                            </li>
                            <li className='navigation__item'>
                                <button className='red-button home-btn'>
                                    <Link to='/' style={{ textDecoration: 'none', color: ' #fff290', padding: '1rem', fontFamily: 'Lemonada, cursive' }}>Home</Link>
                                </button>
                            </li>

                        </div>
                    </ul>
                </nav>
            </div>
        )
    
}
export default NavBar;