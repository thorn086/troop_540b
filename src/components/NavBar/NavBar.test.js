import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import {shallow, configure} from 'enzyme'
import NavBar from './NavBar'

configure({adapter: new Adapter()})

describe('NavBar',()=>{
    const navBar = shallow(<NavBar />)
    it('properly renders', ()=>{
        expect(navBar).toMatchSnapshot()
    })
})