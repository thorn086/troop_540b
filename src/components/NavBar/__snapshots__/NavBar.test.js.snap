// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar properly renders 1`] = `ShallowWrapper {}`;
